### PR TITLE
[Task #5] Implement docker-compose

### PR DESCRIPTION
**Pull request by Aleksander-Drumev-WA**

### CODE REVIEW

File: Pizzeria.Tests/BasketDataServiceTests.cs [hunk 1]
```diff
@@ -216,6 +216,7 @@ public async Task User_changes_basketItem_quantity_successfully()
             var firstCatalogItem = catalogItems.First();
             var dto = new BasketItemDTO
             {
+                Id = 1,
```
Comment (var1ap-WA) [inline Pizzeria.Tests/BasketDataServiceTests.cs]: magic Id

File: WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs [hunk 2]
```diff
@@ -60,25 +66,26 @@ public async Task<int> AddItemToBasketAsync(CatalogItemToBasketItemRequest reque
...
-            var localBasketItem = updatedBasketItem.Adapt<BasketItem>();
-
-            var catalogItem = _dbContext
-                .CatalogItems
-                .First(ci => ci.Id == updatedBasketItem.CatalogItemId);
-
+            var localBasketItem = _dbContext
+            .BasketItems
+            .Include(bi => bi.CatalogItem)
+            .FirstOrDefault(bi => bi.Id == updatedBasketItem.Id);
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs]: Why not async ?

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs [hunk 3]
```diff
@@ -22,15 +22,24 @@ public CatalogDataService(AppDbContext dbContext)
...
         {
-            var catalogItem = dto.Adapt<CatalogItem>();
+            var catalogItem = _dbContext
+                              .CatalogItems
+                              .First(ci => ci.Id == dto.Id);
 
-            if (dto.Id == null)
+            if (catalogItem == null)
             {
+                catalogItem = new CatalogItem
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs]: use mapster instead

File: WA.Pizza.Web/Extensions/ApplicationBuilderExtensions.cs [hunk 4]
```diff
@@ -15,6 +15,8 @@ public static void SeedDatabase(this IApplicationBuilder applicationBuilder)
             using var services = applicationBuilder.ApplicationServices.CreateScope();
 
             var dbContext = services.ServiceProvider.GetService<AppDbContext>();
+            dbContext!.Database.Migrate();
```
Comment (var1ap-WA) [inline WA.Pizza.Web/Extensions/ApplicationBuilderExtensions.cs:18]: correct, your DB must create by this command

File: WA.Pizza.Web/Program.cs [hunk 5]
```diff
@@ -14,13 +14,13 @@
 
 Log.Logger = new LoggerConfiguration()
     .WriteTo.Console()
+    .WriteTo.Seq("http://localhost:5341")
```
Comment (var1ap-WA) [inline WA.Pizza.Web/Program.cs]: hardddcode is bad, use appsettings
and u need call seq by docker compose service name

File: WA.Pizza.Web/Program.cs [hunk 6]
```diff
@@ -66,5 +66,9 @@
     name: "default",
     pattern: "{controller=Home}/{action=Index}/{id?}");
 
-app.SeedDatabase();
+//app.SeedDatabase();
```
Comment (var1ap-WA) [inline WA.Pizza.Web/Program.cs]: uncomment it

File: WA.Pizza.Web/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "AllowedHosts": "*",
+  "ConnectionStrings": {
+    "Default": "Server=localhost,1433;Database=WA.PizzaDB;User=sa;Password=#sql-pass22_"
```
Comment (var1ap-WA) [inline WA.Pizza.Web/appsettings.json]: Server=mssql
as your service name in compose

File: Pizzeria.Tests/BasketDataServiceTests.cs (hunk 1 above)
Comment (Aleksander-Drumev-WA) [reply Pizzeria.Tests/BasketDataServiceTests.cs]: Replaced

File: WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs (hunk 2 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Services/BasketDataService.cs]: It's Async now

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs (hunk 3 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs]: Done

File: WA.Pizza.Web/Program.cs (hunk 5 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Web/Program.cs]: Done

File: WA.Pizza.Web/Program.cs (hunk 6 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Web/Program.cs]: Done

File: WA.Pizza.Web/appsettings.json (hunk 7 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Web/appsettings.json]: Replaced

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs [hunk 8]
```diff
@@ -22,15 +22,18 @@ public CatalogDataService(AppDbContext dbContext)
 
         public async Task<int> AddOrUpdateAsync(CatalogItemDTO dto)
         {
-            var catalogItem = dto.Adapt<CatalogItem>();
+            var catalogItem = _dbContext
+                              .CatalogItems
+                              .First(ci => ci.Id == dto.Id);
```
Comment (var1ap-WA) [inline WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs:27]: FirstOrDefault

**Review by var1ap-WA — APPROVED** [review]

File: WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs (hunk 8 above)
Comment (Aleksander-Drumev-WA) [reply WA.Pizza.Infrastructure/Data/Services/CatalogDataService.cs:27]: Done

**Pull request merged**